application/core/LimeMailer.php

```diff
@@ -823,9 +823,19 @@ public function getTokenReplacements()
         $aTokenReplacements["GLOBALOPTINURL"] = App()->getController()
             ->createAbsoluteUrl("/optin/participants", array("surveyid" => $this->surveyId, "token" => $token,"langcode" => $language));
         $this->addUrlsPlaceholders("GLOBALOPTINURL");
-        $aTokenReplacements["SURVEYURL"] = $survey->getSurveyUrl($language, ["token" => $token]);
+        $surveyUrlCreator = new \LimeSurvey\Models\Services\SurveyUrl($language, ["token" => $token]);
+        $aTokenReplacements["SURVEYURL"] = $surveyUrlCreator->getUrl(
+            $survey->sid,
```

Comment:
Hm, looking at the usage, would it make more sense to pass the entire survey object? Since it's using multiple parts of it anyway.

Reply:
This will take more time. I'm not sure if this is really necessary right now...

Reply:
30 min of work or?

Reply:
> Hm, looking at the usage, would it make more sense to pass the entire survey object? Since it's using multiple parts of it anyway.

We already have a static var for survey->findByPk

Unsure on the optimisation here.